File: OrchidCore/src/main/java/com/eden/orchid/impl/server/admin/lists/ParsersList.java
package com.eden.orchid.impl.server.admin.lists;

import com.eden.orchid.api.compilers.OrchidParser;
import com.eden.orchid.api.server.admin.AdminList;

import javax.inject.Inject;
import java.util.Collection;
import java.util.Set;
import java.util.TreeSet;

public class ParsersList implements AdminList<OrchidParser> {

    private Set<OrchidParser> list;

    @Inject
    public ParsersList(Set<OrchidParser> list) {
        this.list = new TreeSet<>(list);
    }

    @Override
    public String getKey() {
        return "parsers";
    }

    @Override
    public Collection<OrchidParser> getItems() {
        return list;
    }

    @Override
    public String getItemId(OrchidParser item) {
        return item.getClass().getSimpleName();
    }

    @Override
    public OrchidParser getItem(String id) {
        for(OrchidParser item : getItems()) {
            if(id.equals(item.getClass().getSimpleName())) {
                return item;
            }
        }
        return null;
    }
}

File: plugins/OrchidPluginDocs/src/main/kotlin/com/eden/orchid/pluginDocs/lists/ParsersList.kt
package com.eden.orchid.pluginDocs.lists

import com.eden.orchid.api.compilers.OrchidParser
import com.eden.orchid.api.server.admin.AdminList
import java.util.TreeSet
import javax.inject.Inject

@JvmSuppressWildcards
class ParsersList @Inject
constructor(list: Set<OrchidParser>) : AdminList {

    private val list: Set<OrchidParser>

    init {
        this.list = TreeSet(list)
    }

    override fun getKey(): String {
        return "parsers"
    }

    override fun getItems(): Collection<Any> {
        return list
    }

    override fun getItemId(item: Any): String {
        return item.javaClass.simpleName
    }

    override fun getItem(id: String): Any? {
        for (item in items) {
            if (id == item.javaClass.simpleName) {
                return item
            }
        }
        return null
    }
}

